src/ServiceContainer/RestExtension.php

```diff
@@ -51,11 +49,62 @@ public function configure(ArrayNodeDefinition $builder)
      */
     public function load(ContainerBuilder $container, array $config)
     {
-        $guzzleDefinition = new Definition(Guzzle::class, [$config['guzzle']]);
-        $httpClientDefinition = new Definition(GuzzleHttpClient::class, [$guzzleDefinition]);
-        $container->setDefinition('rest.http_client', $httpClientDefinition);
+        $options = is_array($config['guzzle']['config']) ? $config['guzzle']['config'] : [];
+
+        $this->addDiscoveryFactory($container);
+        $this->addGuzzleFactory($container, $config);
+        $this->addArgumentResolver($container, $options);
+    }
 
-        $argumentResolverDefinition = new Definition(HttpClientArgumentResolver::class, [new Reference('rest.http_client')]);
+    /**
+     * @param ArrayNodeDefinition $builder
+     */
+    private function configureGuzzle(ArrayNodeDefinition $builder)
+    {
+        $guzzle = $builder->children()
+            ->arrayNode('guzzle')
+            ->canBeEnabled();
```

Comment:
validation might be useful that triggers an error in case the Guzzle adapter is not present (thus avoiding runtime errors)